### PR TITLE
Add DELETE feature

### PR DESCRIPTION
**Pull request by serpient**
### If you wanted to define the deletion of a specific resource in Application:
```
router.get("/delete_me.txt", (Request request, Response response) -> {
        response.sendFile("/" + "delete_me.txt");
 });

router.delete("/delete_me.txt", (Request request, Response response) -> {
        router.deleteResource(request.getRoute());
        response.successfulDelete();
});
```

### CODE REVIEW

File: src/main/java/application/App.java [hunk 1]
```diff
@@ -39,13 +40,22 @@ private static Router createRouter() {
...
-            response.successfulPost(request.getRoute());
+            response.sendBody(request.getBody().getBytes(), request.getContentFileType());
+            response.setStatus(StatusCode.ok);
         });
 
         app.get("/method_options", (Request request, Response response) -> {});
 
         app.head("/method_options", (Request request, Response response) -> {});
 
+        app.get("/method_options2", (Request request, Response response) -> {});
```
Comment (aaizuss) [inline src/main/java/application/App.java]: What are all of the `/method_options2` endpoints for?

File: src/main/java/http_protocol/MIMETypes.java [hunk 2]
```diff
@@ -8,15 +8,15 @@
...
     public static String png = "image/png";
-
-    private static HashMap<String, String> types = new HashMap<>();
+    private static HashMap<String, String> types = new HashMap<String, String>() {{
+        put(plain, "txt");
+        put(html, "html");
+        put(css, "css");
+        put(jpeg, "jpg");
+        put(png, "png");
+    }};
```
Comment (aaizuss) [inline src/main/java/http_protocol/MIMETypes.java]: Nice, this is better than what you had before

File: src/main/java/application/App.java (hunk 1 above)
Comment (serpient) [reply src/main/java/application/App.java]: this is to complete the acceptance tests from http_server_spec.

File: src/main/java/http_server/Response.java [hunk 3]
```diff
@@ -113,6 +114,10 @@ public void head(byte[] bodyContent, String contentType) {
         setHeader(Headers.contentType, contentType);
     }
 
+    public void successfulDelete() {
```
Comment (aaizuss) [inline src/main/java/http_server/Response.java]: Ultimately, this method is setting an instance variable, and this name does not make that clear. And you already have a public `setStatus` function, so I'm curious why you don't use that when setting the status code for a delete.

**Pull request merged**